src/loo_nonfactorizable.jl

```diff
@@ -0,0 +1,68 @@
+function diag_cov_from_cov_chol(F::Cholesky)
+    if F.uplo == 'U'
+        Uinv = inv(UpperTriangular(F.factors))
+        return map(Base.Fix1(sum, abs2), eachrow(Uinv))
+    else
+        Linv = inv(LowerTriangular(F.factors))
+        return map(Base.Fix1(sum, abs2), eachcol(Linv))
+    end
+end
+
+whitened_residuals(F::Cholesky, r::AbstractVector) = F \ r
+whitened_residuals(F::AbstractMatrix, r::AbstractVector) = F * r
+
+function pointwise_normal_loglikelihood!(
```

Comment:
I prefer to drop the method that takes mean and cov (or `Cholesky`) directly and just keep the one that takes an `MvNormal` or `MvNormalCanon`.

Reply:
Agreed, comment included.